walory-pl.py

```diff
@@ -0,0 +1,252 @@
+                
+            import codecs # import modułu codecs, z ktorego funkcja open pozwala wybrać kodowanie
+            plik = codecs.open (sciezka + "\\" + nazwa_pliku + ".txt",encoding="utf-8") #otwarcie wskazanego pliku
+
+            break
+        except FileNotFoundError:
+            print("Podany plik nie istnieje.")
+        
+    
+    tresc_pliku = plik.readlines() #wczytanie pliku liniami (każda linijka to kolejny element listy)
+    
+    plik.close()
+    
+    wybrany_zestaw = []
+    for i in range(len(tresc_pliku)):# pętla z liczbą powtorzeń rowną liczbie wierszy w pliku, rozdziela kolejne wiersze wg separatora
+        kolejny_wiersz = tresc_pliku[i].split(',')  # podział wiersza wg separatora na listę
+        kolejny_wiersz[1] = float(kolejny_wiersz[1])    # zamiana wspołrzednych z str a float
+        kolejny_wiersz[2] = float(kolejny_wiersz[2])    # jw
+        ### !!!! - poniższe 2 linijki muszą być warunkowe (jesli plik ma te kolumny)
+        if len(kolejny_wiersz) < 4: 
+            kolejny_wiersz = kolejny_wiersz + [i, 5] # dodanie 2 pol: id i pola, ktore będzie wagą w losowaniu
+        else:
+            kolejny_wiersz[3] = int(kolejny_wiersz[3])  
+            kolejny_wiersz[4] = int(kolejny_wiersz[4])
+        wybrany_zestaw.append(kolejny_wiersz) #dodawanie kolejnych wierszy do wybranego zestawu
+            
+    return wybrany_zestaw
+#%%
+def zapis_do_pliku (dane):
+    """
+    Funkcja zapisująca do pliku wyniki uczenia się - w ostatniej kolumnie im większa liczba tym gorzej opanowany materiał i częsciej będzie losowany
+    Zapisuje w kodowaniu uft-8
+    !!!! wprowadzić ewentualnie wpisywanie daty do nazwy
+    !!!! czy dodać opcję wyjscia bez zapisywania wynikow?
+     
+    """
+    import os
+   
+    sciezka = (os.getcwd()) # sprawdzenie w jakim katalogu się znajdujemy (katalog bieżący)
+    
+    nazwa_pliku_out = input("Podaj nazwę pliku do zapisu wynikow, bez rozszerzenia *.txt: ")
+    
+    czy_istnieje = os.path.isfile(sciezka + "\\" + nazwa_pliku_out + ".txt")
+        
+    ### pętla i warunki do weryfikacji czy nie nadpisuje pliku (ew. swiadomego nadpisania)
+    if (czy_istnieje == True):
+        czy_petla = "t"
+    else: czy_petla = "n"
+    
+    while czy_petla == "t":
+        czy_nadpisac = input("Plik o podanej nazwie istnieje. Czy nadpisać? t/n: ")
+        if czy_nadpisac != "t":
+            nazwa_pliku_out = input("Podaj INNĄ nazwę pliku do zapisu wynikow, bez rozszerzenia *.txt: ")
+        else: czy_petla = "n"
+
+    print("Plik wybrany do zapisu: " + sciezka + "\\" + nazwa_pliku_out + ".txt")
+    ### kod otwierający plik       
+    import codecs 
+    plik = codecs.open(sciezka + "\\" + nazwa_pliku_out + ".txt", 'w',encoding="utf-8") # - w oznacza, że będzie zapisywany plik (ewentualny istniejący zostanie nadpisany)
+    
+    
+    for i in range(len(dane)-1): #pętla powtarza dla liczby wierszy minus 1 (bo ostatni ma być bez \n)
+        wiersz = ""
```

Comment:
Wydajniej jest, jeśli chcemy gromadzić dane, robić to w liście, a na końcu dopiero przerobić tę listę na string poleceniem `join`. Ale w tym przypadku w ogóle lepiej użyć `csv` z biblioteki standardowej.